src/app/src/pages/eval/components/ResultsView.tsx

```diff
@@ -174,9 +174,14 @@ export default function ResultsView({
   const { showToast } = useToast();
   const [searchText, setSearchText] = React.useState(searchParams.get('search') || '');
   const [debouncedSearchText] = useDebounce(searchText, 200);
-  const handleSearchTextChange = React.useCallback((text: string) => {
-    setSearchText(text);
-  }, []);
+  const handleSearchTextChange = React.useCallback(
+    (text: string) => {
+      const searchEntries = Object.fromEntries(searchParams.entries());
+      setSearchParams({ ...searchEntries, search: text });
```

Comment:
You can do `setSearchParams((prev) => ({ ...prev, search: text }));` and get rid of `searchEntities`.